OpenRA.Game/Network/LocalizedMessage.cs

```diff
@@ -0,0 +1,130 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Fluent.Net;
+using OpenRA.Widgets;
+
+namespace OpenRA.Network
+{
+	[Flags]
+	public enum SupportedTypes
```

Comment:
The `SupportedTypes` and `Argument` type names are far too general to expose directly in the `OpenRA.Network` namespace. Something like `LocalizedMessageArgumentType` and `LocalizedMessageArgument` would be better (but i'm not insisting on these specific names if you have better ideas).

Reply:
Or these could alternatively be moved inside the `LocalizedMessage` definition, meaning that they would be accessed from external code (does anything even do this?) as `LocalizedMessage.Argument` etc.

Reply:
I contained this a bit now.